pkg/exporter/syslogexporter/exporter_test.go

```diff
@@ -24,3 +69,104 @@ func TestInitExporter(t *testing.T) {
+	exp, err := initExporter(cfg, createExporterCreateSettings())
+	require.NoError(t, err)
+	return exp, err
+}
+
+func buildInvalidExporter(t *testing.T, server net.TCPListener, cfg *Config) (*syslogexporter, error) {
+	var port string
+	var err error
+	hostPort := server.Addr().String()
+	cfg.Endpoint, port, err = net.SplitHostPort(hostPort)
+	require.NoError(t, err, "could not parse endpoint")
+	require.NotNil(t, port)
+	invalidPort := "112" // Assign invalid port
+	cfg.Port, err = strconv.Atoi(invalidPort)
+	require.NoError(t, err, "type error")
+	exp, err := initExporter(cfg, createExporterCreateSettings())
+	require.NoError(t, err)
+	return exp, err
+}
+
+func createServer() (net.TCPListener, error) {
+	var addr net.TCPAddr
+	addr.IP = net.IP{127, 0, 0, 1}
+	addr.Port = 0
+	testServer, err := net.ListenTCP("tcp", &addr)
+	return *testServer, err
+}
+
+func prepareExporterTest(t *testing.T, cfg *Config, invalidExporter bool) *exporterTest {
+	// Start a test syslog server
+	testServer, err := createServer()
+	require.NoError(t, err, "failed to start test syslog server")
+	if invalidExporter {
+		exp, err := buildInvalidExporter(t, testServer, cfg)
+		require.NoError(t, err, "Error building exporter")
+		require.NotNil(t, exp)
+		return &exporterTest{
+			srv: testServer,
+			exp: exp,
+		}
+	}
+	exp, err := buildValidExporter(t, testServer, cfg)
+	require.NoError(t, err, "Error building exporter")
+	require.NotNil(t, exp)
+	require.NoError(t, err)
+	return &exporterTest{
+		srv: testServer,
+		exp: exp,
+	}
+
+}
+
+func createTestConfig() *Config {
+	config := createDefaultConfig().(*Config)
+	config.Protocol = "tcp"
+	config.TLSSetting.Insecure = true
+	return config
+}
+
+func TestSyslogExportSuccess(t *testing.T) {
+	test := prepareExporterTest(t, createTestConfig(), false)
+	require.NotNil(t, test.exp)
+	defer test.srv.Close()
+	go func() {
+		buffer := exampleLog(t)
+		logs := LogRecordsToLogs(buffer)
+		err := test.exp.pushLogsData(context.Background(), logs)
+		require.NoError(t, err, "could not send message")
+	}()
+	err := test.srv.SetDeadline(time.Now().Add(time.Second * 1))
+	require.NoError(t, err, "cannot set deadline")
+	conn, err := test.srv.AcceptTCP()
+	require.NoError(t, err, "could not accept connection")
+	defer conn.Close()
+	b, err := io.ReadAll(conn)
+	require.NoError(t, err, "could not read all")
+	assert.Equal(t, string(b), expectedForm)
+}
+
+func TestSyslogExportFail(t *testing.T) {
+	test := prepareExporterTest(t, createTestConfig(), true)
+	defer test.srv.Close()
+	buffer := exampleLog(t)
+	logs := LogRecordsToLogs(buffer)
+	consumerErr := test.exp.pushLogsData(context.Background(), logs)
+	err := test.srv.SetDeadline(time.Now().Add(time.Second * 1))
+	require.NoError(t, err, "cannot set deadline")
+	conn, err := test.srv.AcceptTCP()
+	require.ErrorContains(t, err, "i/o timeout")
+	require.Nil(t, conn)
+	assert.ErrorContains(t, consumerErr, "dial tcp 127.0.0.1:112: connect")
+}
```

Comment:
We should check if `consumerErr` contains unsent logs

Reply:
it looks like one cannot return failed logs from the documentation `NewLogs` just returns an `error`